design/KruizeDatabaseDesign.md

```diff
@@ -0,0 +1,824 @@
+   </td>
+  </tr>
+  <tr>
+   <td>target_cluster
+   </td>
+   <td>string
+   </td>
+   <td>The target Kubernetes cluster for the application. This can be "local" or "remote".
+   </td>
+  </tr>
+  <tr>
+   <td>datasource
+   </td>
+   <td>array of json
+   </td>
+   <td>The data sources used to fetch metrics results.
+   </td>
+  </tr>
+  <tr>
+   <td>extended_data
+   </td>
+   <td>json
+   </td>
+   <td>Additional data for the experiment in JSON format, which includes complete Kubernetes objects details, recommendation settings, trial settings, and ACM additional settings.
+   </td>
+  </tr>
+  <tr>
+   <td>meta_data
+   </td>
+   <td>json
+   </td>
+   <td>Additional metadata for the experiment in JSON format.
+   </td>
+  </tr>
+</table>
+
+### Kruize entity related API's
+
+The following REST API's are available for the kruize_experiment entity:
+
+#### **1. CreateExperiment**
+
+This API creates a new experiment using POST method and JSON data. The following attributes are required to create an
+experiment record:
+
+* version
+* experiment_name
+* cluster_name
+* performance_profile
+* mode
+* target_cluster
+* datasource
+* extended_data
+
+##### Request Method
+
+HTTP_METHODE: POST
+
+##### Rest API Curl
+
+```
+ curl --location --request POST 'http://127.0.0.1:8080/createExperiment' \
+			--header 'Content-Type: application/json' \
+			--data-raw '[
+			    {
+				"version": "1.0",
+				"experiment_name": "quarkus-resteasy-autotune-min-http-response-DB10",
+				"cluster_name": "cluster-one-division-bell",
+				"performance_profile": "resource-optimization-openshift",
+				"mode": "monitor",
+				"target_cluster": "remote",
+				"kubernetes_objects": [
+				    {
+					"type": "deployment",
+					"name": "tfb-qrh-deployment",
+					"namespace": "default",
+					"containers": [
+					    {
+						"container_image_name": "kruize/tfb-db:1.15",
+						"container_name": "tfb-server-0"
+					    },
+					    {
+						"container_image_name": "kruize/tfb-qrh:1.13.2.F_et17",
+						"container_name": "tfb-server-1"
+					    }
+					]
+				    }
+				],
+				"trial_settings": {
+				    "measurement_duration": "15min"
+				},
+				"recommendation_settings": {
+				    "threshold": "0.1"
+				}
+			    }
+			]'
+
+```
+
+##### Db query
+
+```
+insert into kruize_experiment (cluster_name, experiment_name, extended_data, meta_data, mode,  performance_profile, status, target_cluster, datasource, version) values (?, ?, ?, ?, ?, ?, ?, ?, ?)
+
+```
+
+#### **2. UpdateExperiment**
+
+The purpose of this API is to facilitate the modification of various aspects of an experiment, such as replacing
+Kubernetes objects, adding or removing deployment/statefulset types, adding or removing containers, updating the
+experiment's status, and adjusting settings such as trial and recommendation settings.
+
+##### Request Method
+
+HTTP_METHODE: POST
+
+##### Rest API Curl
+
+###### add new container
```

Comment:
Instead of saying add a new container / delete container, would suggest that this "Updates the kubernetes objects". In the subtext, you can mention that this can be used to add / delete containers